src/nvim/drawline.c

```diff
@@ -2967,16 +2967,17 @@ int win_line(win_T *wp, linenr_T lnum, int startrow, int endrow, bool nochange,
         wlv.need_showbreak = true;
       }
       if (statuscol.draw) {
-        if (wlv.row == startrow + wlv.filler_lines + 1
-            || wlv.row == startrow + wlv.filler_lines) {
-          // Re-evaluate 'statuscolumn' for the first wrapped row and non filler line
-          statuscol.textp = NULL;
-        } else if (statuscol.textp) {
+        if (wlv.row == startrow + wlv.filler_lines) {
+          statuscol.textp = NULL;  // re-evaluate for first non-filler line
+        } else if (vim_strchr(p_cpo, CPO_NUMCOL) && wlv.row > startrow + wlv.filler_lines) {
+          statuscol.draw = false;  // don't draw status column if "n" is in 'cpo'
```

Comment:
Hmm, sign and fold columns are still needed on wrapped screen lines with `cpo-n`.

Reply:
This is for `cpo+n` (the old comment was incorrect).

Reply:
Yes. They are still needed.

Reply:
Yes, they are drawn by falling back the default columns by setting `statuscol.draw` to false.

Reply:
Or should be, did you observe that this is not the case?

Reply:
> Yes, they are drawn by falling back the default columns by setting `statuscol.draw` to false.

Hmm, won't that look a bit inconsistent?

Reply:
It might, this was the compromise I decided on in the original PR.

Reply:
The alternative is placing the burden of making it look right on the user when `v:virtnum > 0`.

Reply:
But that would not even be possible, unless if we don't pad the statuscolumn with whitespace for wrapped lines and `cpo+=n`.

Currently a shorter returned string would simply be padded to the maximum statuscolumn width, nullifying `cpo+=n`.